src/gluonts/trainer/model_averaging.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+import glob
+import json
+
+# Third-party imports
+import mxnet as mx
+import numpy as np
+
+EPOCH_INFO_STRING = "epoch-info"
+
+
+def save_epoch_info(tmp_path, epoch_info):
+    """
+    Writes the current epoch information into a json file in the model path.
+    :param epoch_info: Dictionary, containing with epoch information.
+    :return: None
+    """
+    with open("{}-{}.json".format(tmp_path, EPOCH_INFO_STRING), "w") as f:
+        json.dump(epoch_info, f)
+
+
+def average_parameters(
+    model_path, num_models=5, metric="score", maximize=False, weight="average"
+):
+    """
+    Averages model parameters of serialized models based on the selected model strategy and metric.
+    IMPORTANT: Depending on the metric the user might want to minimize or maximize. The maximize flag has to be
+    chosen appropriately to reflect this.
+
+    :param model_path: str, path to the models directory.
```

Comment:
Could you convert all these docstrings to numpy-style? You can configure PyCharm to automatically style do strings like that when you open triple-quotes